lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -611,6 +611,192 @@ def postprocess_element(self, elem: Element) -> Element:
         return elem
 
 
+class HeaderAugmenterMerger(ElementMerger):
+    """
+    The ``HeaderAugmenterMerger`` groups together different elements in a Document and enhances the text
+    representation of the elements by adding the preceeding section-header/title.
+
+    - It merges elements ("Text", "List-item", "Caption", "Footnote", "Formula" , "Page-footer", "Page-header").
+    - It merges consectuive ("Section-header", "Title") elements.
+    - It adds the preceeding section-header/title to the text representation of the elements (including tables/images).
+    """
+
+    def __init__(self, tokenizer: Tokenizer, max_tokens: int, merge_across_pages: bool = True):
+        self.tokenizer = tokenizer
+        self.max_tokens = max_tokens
+        self.merge_across_pages = merge_across_pages
+
+    def preprocess_element(self, element: Element) -> Element:
+        if element.type == "Image" and "summary" in element.properties and "summary" in element.properties["summary"]:
+            element.data["token_count"] = len(self.tokenizer.tokenize(element.properties["summary"]["summary"] or ""))
+        else:
+            element.data["token_count"] = len(self.tokenizer.tokenize(element.text_representation or ""))
+        return element
+
+    def postprocess_element(self, element: Element) -> Element:
+        del element.data["token_count"]
+        return element
+
+    def merge_elements(self, document: Document) -> Document:
+        """Use self.should_merge and self.merge to greedily merge consecutive elements.
+        If the next element should be merged into the last 'accumulation' element, merge it.
+
+        Args:
+            document (Document): A document with elements to be merged.
+
+        Returns:
+            Document: The same document, with its elements merged
+        """
+        if len(document.elements) < 2:
+            return document
+
+        firstHeader = False
```

Comment:
To keep the naming convention consistent, can you name this `first_header` instead?

Reply:
done